### PR TITLE
Simplify SGSSB Code

### PR DESCRIPTION
**Pull request by Spandamn**
This basically creates a status effect for every SSB mon, and adds it to the mon on switch in. This will help a lot if innate abilities are to be implemented.

### CODE REVIEW

**Review by Lord-Haji — APPROVED** [review]
This is a great idea

**Comment by Spandamn** [comment]
bump

File: mods/sgssb/abilities.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		id: 'primalsurge',
+		onStart: function (source) {
+			this.setWeather('primordialsea');
+			this.setTerrain('electricterrain');
+			this.terrainData.duration = 0;
+		},
+		onAnySetWeather: function (target, source, weather) {
+			if (this.getWeather().id === 'primordialsea' && !(weather.id in {desolateland:1, primordialsea:1, deltastream:1})) return false;
+		},
+		onAnySetTerrain: function (target, source, terrain) {
```
Comment (HoeenCoder) [inline mods/sgssb/abilities.js]: Yeah this event dosen't exist, I tried to add one to fix a bug with seeds but Zarel had a better way to do it.

Comment (Spandamn) [reply mods/sgssb/abilities.js]: oh, then this will need modification to `setTerrain`. Will do that

**Pull request merged**